collect_app/src/main/java/org/odk/collect/android/utilities/AuditEventLogger.java

```diff
@@ -139,13 +142,23 @@ boolean isDuplicateOfLastAuditEvent(AuditEvent.AuditEventType eventType) {
      * Exit a question
      */
     public void exitView() {
+        exitView(null);
+    }
+
+    public void exitView(Map<FormIndex, IAnswerData> answers) {
         if (isAuditEnabled()) {
             // Calculate the time and add the event to the auditEvents array
             long end = getEventTime();
             for (AuditEvent aev : auditEvents) {
                 if (!aev.isEndTimeSet() && aev.isIntervalAuditEventType()) {
                     addLocationCoordinatesToAuditEventIfNeeded(aev);
                     aev.setEnd(end);
+                    if (answers != null && !answers.isEmpty() && aev.getAuditEventType() == AuditEvent.AuditEventType.QUESTION) {
+                        Map.Entry entry = answers.entrySet().iterator().next();
```

Comment:
This means the answers from questions in a field list will be provided in some arbitrary order because the map is a hashmap, right?

Reply:
Yeah, the order should be the same as questions on the screen.

Reply:
Ah, right, this is actually worse than I had originally imagined, I think. Isn't it the case that the audit event has a reference and that the answer that gets associated with that audit event might not match the reference to the question? How about passing a map from reference to answer rather than index to answer? By reference I mean what is stored in the `AuditEvent.node` field (e.g. `/data/name`), so that you can just match the event to its answer?